src/viewProviders/resources.ts

```diff
@@ -439,6 +452,17 @@ export class ResourceViewProvider implements vscode.TreeDataProvider<ResourceVie
       }
     });
   }
+
+  /** Update internal state when the search string is set or unset. */
+  setSearch(searchString: string | null): void {
```

Comment:
This doesn't have anything to do with the total count / usage event changes in this PR, but the Topics/Schemas views already incorporated this kind of method and it makes things easier for this view as well (while also providing more justification for moving toward a central base view provider class).